### PR TITLE
Fix columnar menu completion corrupting the editor insertion point

### PR DESCRIPTION
**Pull request by csko**
See: https://github.com/nushell/nushell/issues/7885 for more context.

Repro:
1. type `` open FILENAME` ``
2. press TAB (`` ` `` gets removed at the end)
3. press `` ` ``

### CODE REVIEW

**Comment by csko** [comment]
Should we bump the version?

File: src/menu/columnar_menu.rs [hunk 1]
```diff
@@ -667,7 +667,8 @@ impl Menu for ColumnarMenu {
             line_buffer.replace_range(start..end, &value);
 
             let mut offset = line_buffer.insertion_point();
-            offset += value.len().saturating_sub(end.saturating_sub(start));
+            offset += value.len();
+            offset -= end.saturating_sub(start);
```
Comment (sholderbach) [inline src/menu/columnar_menu.rs:671]: Can we prove that this never underflows? (Not sure if we have enough tests with gnarly insertion points in compleitions in the `reedline` repo for that)

File: src/menu/columnar_menu.rs [hunk 2]
```diff
@@ -810,4 +811,25 @@ mod tests {
             append_whitespace: false,
         }
     }
+
+    #[test]
+    fn test_menu_replace_backtick() {
+        // https://github.com/nushell/nushell/issues/7885
```
Comment (sholderbach) [inline src/menu/columnar_menu.rs:817]: Thanks for the practical test!

**Review by sholderbach — COMMENTED** [review]
Thanks for tackling this problem! 

The menu system at the moment is still overly complex in my view. Thanks for spotting that bug!

File: src/menu/columnar_menu.rs (hunk 1 above)
Comment (csko) [reply src/menu/columnar_menu.rs]: good point, I've changed these to saturating_sub/add

**Review by sholderbach — APPROVED** [review]
Let's give this a go

**Pull request merged**